truss/util/path.py

```diff
@@ -1,18 +1,39 @@
+import fnmatch
 import os
 import random
 import string
 import tempfile
 from contextlib import contextmanager
-from distutils.dir_util import copy_tree, remove_tree
+from distutils.dir_util import remove_tree
 from distutils.file_util import copy_file
 from pathlib import Path
 from typing import List, Optional, Tuple, Union
 
 from truss.patch.hash import str_hash_str
 
+# .truss_ignore is a fixed file in the Truss library that is used to specify files
+# that should be ignored when copying a directory tree such as .git directory.
+FIXED_TRUSS_IGNORE_PATH = Path(__file__).parent / ".truss_ignore"
 
-def copy_tree_path(src: Path, dest: Path) -> List[str]:
-    return copy_tree(str(src), str(dest), verbose=0)
+
+def copy_tree_path(src: Path, dest: Path) -> None:
+    """Copy a directory tree, ignoring files specified in .truss_ignore."""
+    patterns = load_trussignore_patterns(FIXED_TRUSS_IGNORE_PATH)
+
+    if not dest.exists():
+        dest.mkdir(parents=True)
+
+    for sub_path in src.rglob("*"):
+        if is_ignored(sub_path, patterns, base_dir=src):
```

Comment:
aside from this ignore path, is this the same implementation as the copy_tree from before. Just curious. Would be good to have a comment

Reply:
in practice, yes this exhibits the same behavior as `copy_tree`. the changes however are 

- `copy_tree` from before returns a list of output files. i made sure to check that we never use the output anywhere so the signature of the function has changed 
- `copy_tree` provides some extra args that affect how the copied file should be copied (i.e preserving some metadata) but we can take advantage of these features, if we want, via the `copy_file`